### DISCUSSION TITLE
No .json results in "allure-results" folder

### DISCUSSION THREAD

**Discussion by JAVA-NOVACHOK** [mmisty/cypress-allure-adapter#97] [General]
Hello. I am facing a problem with getting the test results. After the test run folder "allure-results" has no "*-result.json" files. There are only files like "categories.json" and "environment.properties" because of "before:run".
What can be the cause of this?
`

    e2e: {
    experimentalRunAllSpecs: true,
    testIsolation: false,
    async setupNodeEvents(cypressOn, config) {
      const on = require('cypress-on-fix')(cypressOn)
      const reporter = configureAllureAdapterPlugins(on, config);
      grep(config)
      cypressSplit(on, config)
      on('before:run', details => {
        reporter?.writeEnvironmentInfo({
            info: {
                os: details.system.osName,
            osVersion: details.system.osVersion,
            browser: details.browser?.displayName + ' ' + details.browser?.version,
            ...config.env
          },
        });
      
         reporter?.writeCategoriesDefinitions({ categories: './allure-categories/categories.json' });
      });
`

**Comment by mmisty** [comment]
@JAVA-NOVACHOK Hello, 
did you do setup like it says [here](https://github.com/mmisty/cypress-allure-adapter/blob/main/README.pack.md#1-update-support) ? 
If yes, could you please attach your:
 - support file
 - cypress.config.ts file
 - package.json
 
so I can understand where is the problem

**Comment by JAVA-NOVACHOK** [comment]
I did everything according to the installation process. But maybe I missed something. 
For files "e2e" and "cypress.config" I changed format to be able to attach them here (.js -> .json)
[cypress.config.json](https://github.com/mmisty/cypress-allure-adapter/files/14109062/cypress.config.json)
[package.json](https://github.com/mmisty/cypress-allure-adapter/files/14109063/package.json)
[e2e.json](https://github.com/mmisty/cypress-allure-adapter/files/14109064/e2e.json)

**Comment by JAVA-NOVACHOK** [comment]
I did this to be able to attach the file. But in the project it is .js

**Comment by mmisty** [comment]
@JAVA-NOVACHOK 
I can see one suspicious thing which may cause your problem -  you  do not return config in setupNodeEvents  `cypress.config`
Also suggesting to check out [setup examples](https://github.com/mmisty/cypress-allure-adapter-example/tree/main/examples)

**Reply by JAVA-NOVACHOK** [reply]
@mmisty Hello. Please advise according to my comment below.https://github.com/mmisty/cypress-allure-adapter/discussions/97#discussioncomment-8333493

**Comment by JAVA-NOVACHOK** [comment]
In "setupNodeEvents()" I return the configuration object. It is not "config" but "...config" is like inside this object. Maybe plugin does not recognize or cannot map the values of this configurations?
![config](https://github.com/mmisty/cypress-allure-adapter/assets/44897473/c6372bed-e7e4-437d-a75f-08d31147d159)

**Reply by mmisty** [reply]
@JAVA-NOVACHOK  I see that you overriding env variable in config object, so your `allure` variable is unset 
Please try adding : 
```javascript
{
  ...config, 
  //...
  env: {
   ...config.env
    // your overrides
   }
}
```

**Reply by JAVA-NOVACHOK** [reply]
I just figured this out and your answer confirmed my assumption. It worked. Thanks.
